### PR TITLE
LPS-144040 Allow to publish workflow definition

### PR DESCRIPTION
**Pull request by naraandrade**
Story: [LPS-129928](https://issues.liferay.com/browse/LPS-129928)

### CODE REVIEW

**Comment by naraandrade** [comment]
ci:test:relevant

**Comment by ramonmsp** [comment]
Just started reviewing :)

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 1]
```diff
@@ -60,118 +71,233 @@ export default function UpperToolbar({
...
+					description: '',
+					name: definitionTitle,
+					version,
+				},
+				elements.filter(isNode),
+				elements.filter(isEdge)
+			);
+		}
+
+		fetch(`${baseURL}/workflow-definitions/save`, {
```
Comment (ramonmsp) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:97]: @naraandrade What do you think about single responsability in this function? Maybe this inside a 'saveDefinitionService' function (or any other other name) is more readable.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 2]
```diff
@@ -60,118 +71,233 @@ export default function UpperToolbar({
...
+				description: '',
+				name: definitionTitle,
+				version,
+			},
+			elements.filter(isNode),
+			elements.filter(isEdge),
+			true
+		);
+
+		fetch(`${baseURL}/workflow-definitions/deploy`, {
```
Comment (ramonmsp) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:144]: same here

**Review by ramonmsp — CHANGES_REQUESTED** [review]

**Comment by jotascript** [comment]
Just started reviewing :)

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js (hunk 1 above)
Comment (naraandrade) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:97]: @ramonmsp Thanks for pointing this out :) I'll do it 👍🏻

**Review by ramonmsp — APPROVED** [review]

**Comment by naraandrade** [comment]
ci:test:sf

**Comment by naraandrade** [comment]
ci:test:sf

**Review by jotascript — APPROVED** [review]

**Comment by naraandrade** [comment]
ci:forward

**Comment by naraandrade** [comment]
ci:forward

**Pull request closed without merging**